File: samples/AspireWithJavaScript/AspireJavaScript.AppHost/AppHost.cs
var builder = DistributedApplication.CreateBuilder(args);

var weatherApi = builder.AddProject<Projects.AspireJavaScript_MinimalApi>("weatherapi")
    .WithExternalHttpEndpoints();

builder.AddJavaScriptApp("angular", "../AspireJavaScript.Angular", runScriptName: "start")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .PublishAsDockerFile();

builder.AddJavaScriptApp("react", "../AspireJavaScript.React", runScriptName: "start")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithEnvironment("BROWSER", "none") // Disable opening browser on npm start
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .PublishAsDockerFile();

builder.AddJavaScriptApp("vue", "../AspireJavaScript.Vue")
    .WithRunScript("start")
    .WithNpm(installCommand: "ci")
    .WithReference(weatherApi)
    .WaitFor(weatherApi)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .PublishAsDockerFile();

var reactVite = builder.AddViteApp("reactvite", "../AspireJavaScript.Vite")
    .WithReference(weatherApi)
    .WithEnvironment("BROWSER", "none");

// Bundle the output of the Vite app into the wwwroot of the weather API
weatherApi.PublishWithContainerFiles(reactVite, "./wwwroot");

builder.Build().Run();

File: samples/aspire-with-javascript/ts/apphost.ts
import { createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const weatherApi = builder.addProject("weatherapi", "../AspireJavaScript.MinimalApi/AspireJavaScript.MinimalApi.csproj", "https")
    .withExternalHttpEndpoints();

builder.addJavaScriptApp("angular", "../AspireJavaScript.Angular", { runScriptName: "start" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

builder.addJavaScriptApp("react", "../AspireJavaScript.React", { runScriptName: "start" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withEnvironment("BROWSER", "none")
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

builder.addJavaScriptApp("vue", "../AspireJavaScript.Vue")
    .withRunScript("start")
    .withNpm({ installCommand: "ci" })
    .withReference(weatherApi)
    .waitFor(weatherApi)
    .withHttpEndpoint({ env: "PORT" })
    .withExternalHttpEndpoints()
    .publishAsDockerFile();

const reactVite = builder.addViteApp("reactvite", "../AspireJavaScript.Vite")
    .withReference(weatherApi)
    .withEnvironment("BROWSER", "none");

weatherApi.publishWithContainerFiles(reactVite, "./wwwroot");

await builder.build().run();
